presto-spi/src/main/java/com/facebook/presto/spi/eventlistener/EventListener.java

```diff
@@ -19,6 +19,10 @@ default void queryCreated(QueryCreatedEvent queryCreatedEvent)
     {
     }
 
+    default void queryUpdate(QueryUpdateEvent queryUpdateEvent)
```

Comment:
nit `queryUpdated`. Curious: what's the difference between query updated vs query created? So we will have multiple events for presto on spark? If it's the case, can we add java doc to this class?

Reply:
QueryCreated event is generated immediately as soon the query is received but before it is planned. So, it captures information like start time, query_text, session_properties etc. `QueryUpdated` event will be generating any intermediate events while the query is in progress. We can use it for seeing query_plan, partial stats (once AQE is in) and other information which can be useful to capture and show while query is running. And once query finishes, queryCompleted event will update all the metrics and query_plan with stats.